rfcs/geometric-primitives.md

```diff
@@ -0,0 +1,162 @@
+- I still haven't found a case where you would use a `Primitive3d` without needing this positional information that fully defines the primitive in space. If this holds true, it means that storing the positional data inside the primitive is _not_ a waste of cache, which is normally why you would want to separate the transform into a separate component.
+
+#### CPU Efficiency
+
+- Storing the primitive's positional information internally serves as a form of memoization.
+- Because you need the primitive to be fully defined in world space to run useful operations, this means that with a `GlobalTransform` you would need to apply the transform to the primitive every time you need to use it.
+- By applying this transformation only once (e.g. during transform propagation), we only need to do this computation a single time.
+
+#### Ergonomics
+
+- As I've already mentioned a few times, primitives need to be fully defined in world space to do anything useful with them.
+- By making the primitive components fully defined and standalone, computing operations is as simple as: `primitive1.some_function(primitive_2)`, instead of also having query and pass in 2 `GlobalTransform`s in the correct order.
+
+#### Use with Transforms
+
+- For use cases such as oriented bounding boxes, a primitive should be defined relative to its parent.
+- In this case, the primitive would still be fully defined internally, but we would need to include primitive updates analogous to the transform propagation system.
+- For example, a bounding sphere entity would be a child of a mesh, with a `Sphere` primitive and a `Transform`. On updates to the parent's `GlobalTransform`, the bounding sphere's `Transform` component would be used to update the `Sphere`'s position and radius by applying the scale and translation to a unit sphere. This could be applied to all primitives, with the update system being optimized for each primitive.
+
+- What other designs have been considered and what is the rationale for not choosing them?
+- What is the impact of not doing this?
+- Why is this important to implement as a feature of Bevy itself, rather than an ecosystem crate?
+
+## \[Optional\] Prior art
+
+Unity `PrimitiveObjects`: https://docs.unity3d.com/Manual/PrimitiveObjects.html
+Godot `PrimitiveMesh`: https://docs.godotengine.org/en/stable/classes/class_primitivemesh.html#class-primitivemesh
+
+These examples intermingle primitive geometry with the meshes themselves. This RFC makes these distinct.
+
+
+## Unresolved questions
+
+- What parts of the design do you expect to resolve through the RFC process before this gets merged?
+- What parts of the design do you expect to resolve through the implementation of this feature before the feature PR is merged?
+
+### Out of Scope
+
+- Value types, e.g. float vs. fixed is out of scope. This RFC is focused on the core geometry types and is intended to use Bevy's common rendering value types such as `f32`.
+
+## Future possibilities
```

Comment:
Is raycasting impacted by this?

Reply:
I don't think so, do you have something in mind?

Reply:
IIRC this comment was simply about the fact that raycasting was missing from this list at the time I commented.